### DISCUSSION TITLE
v-if does not fix object might be null error on @click

### DISCUSSION THREAD

**Discussion by danielmoessner** [vuejs/language-tools#1351] [Q&A]
I've got the following error:

![image](https://user-images.githubusercontent.com/38765196/170359726-8c979d55-affd-4531-bee0-c75aac3c70d2.png)

Is this an error with Volar? Or am I doing something wrong? Or is it not possible to have TypeScript not complain here?

Code:
```
<template>
  <div v-if="!!myObject">
    <button @click="test(myObject)">Download</button>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from "vue";

interface MyInterface {
  name: string;
}

export default defineComponent({
  setup() {
    const myObject = ref<MyInterface | null>(null);

    const test = (item: MyInterface) => {
      console.log(item.name);
    };

    return {
      myObject,
      test,
    };
  },
});
</script>
```

**Comment by pikax** [comment]
Solved: https://github.com/johnsoncodehk/volar/issues/1249

**Reply by danielmoessner** [reply]
Thanks for your answer. Has it been released yet?

**Reply by johnsoncodehk** [reply]
@danielmoessner yes, you should setup `{ "vueCompilerOptions": { "experimentalAllowTypeNarrowingInInlineHandlers": true } }` in tsconfig.

**Reply by danielmoessner** [reply]
Interesting, thanks. I'll check it out.

**Comment by danielmoessner** [comment]
Since version 1.0.x it is not working anymore. It used to work below 1.x
Do you know more? @johnsoncodehk @pikax

**Reply by johnsoncodehk** [reply]
This option renamed to `narrowingTypesInInlineHandlers`, see https://github.com/johnsoncodehk/volar/blob/master/CHANGELOG.md#100-rc0-2022105.
